### PR TITLE
Ignore frozen_metadata.txt while traversing shadow directory from system.remote_data_paths

### PR DESCRIPTION
**Pull request by aalexfvk**
When executing 
`select * from system.remote_data_paths SETTINGS traverse_shadow_remote_data_paths=1`
we can get a parsing error, because `frozen_metadata.txt` (zero-copy replication related) files in frozen parts do not have a object disk metadata format.

```
2024.10.10 10:14:14.842950 [ 3285 ] {d82c32cb-72b6-46a8-9de1-ef70f880acc0} <Error> executeQuery: Code: 27. DB::Exception: Cannot parse input: expected '\t' before: '%Enet\ndefaul\n': While executing SystemRemoteDataPaths. (CANNOT_PARSE_INPUT_ASSERTION_FAILED) (version 24.8.4.13 (official build)) (from [::ffff:10.1.0.73]:58040) (in query:  SELECT obj_path, obj_size FROM _system.listing_objects_from_object_storage AS object_storage LEFT ANTI JOIN remoteSecure('hosts', system.remote_data_paths) AS object_table ON object_table.remote_path = object_storage.obj_path AND object_table.disk_name = 'object_storage' SETTINGS traverse_shadow_remote_data_paths=1 FORMAT TabSeparated ), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000daffc3b
1. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000007e59fcc
2. DB::Exception::Exception<String&>(int, FormatStringHelperImpl<std::type_identity<String&>::type>, String&) @ 0x0000000007e83d0b
3. DB::throwAtAssertionFailed(char const*, DB::ReadBuffer&) @ 0x000000000db78809
4. DB::DiskObjectStorageMetadata::deserialize(DB::ReadBuffer&) @ 0x00000000114360b6
5. DB::MetadataStorageFromDisk::readMetadataUnlocked(String const&, std::shared_lock<DB::SharedMutex>&) const @ 0x000000001143c4cc
6. DB::MetadataStorageFromDisk::getStorageObjects(String const&) const @ 0x000000001143cd1f
7. DB::SystemRemoteDataPathsSource::generate() @ 0x000000001081e1ed
8. DB::ISource::tryGenerate() @ 0x000000001303219b
9. DB::ISource::work() @ 0x0000000013031ea7
10. DB::ExecutionThreadContext::executeTask() @ 0x000000001304b0e7
11. DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000001303f9b0
12. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::PipelineExecutor::spawnThreads()::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x000000001304106e
13. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false, true>, void*>) @ 0x000000000dbd4649
14. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000dbd8791
15. void* std::__thread_proxy[abi:v15007]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000dbd7509
16. ? @ 0x0000725a20a94ac3
17. ? @ 0x0000725a20b26850
```

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Ignore frozen_metadata.txt while traversing shadow directory from system.remote_data_paths

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit b1e7a4190b20c928013ba21f00621442fbc2c3c6 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70590/b1e7a4190b20c928013ba21f00621442fbc2c3c6/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70590/b1e7a4190b20c928013ba21f00621442fbc2c3c6/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by aalexfvk** [comment]
Fixed documentation. I hope I made it clearer.

@kssenii it would be great to backport this small change.

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -9376,7 +9376,7 @@ Type: Bool
 
 Default value: 0
 
-Traverse shadow directory when query system.remote_data_paths
+Traverse frozen data(shadow directory) in addition to actual table data when query system.remote_data_paths
```
Comment (kssenii) [inline docs/en/operations/settings/settings.md:9379]:
```suggestion
Traverse frozen data (shadow directory) in addition to actual table data when query `system.remote_data_paths`.
```

File: src/Core/Settings.cpp [hunk 2]
```diff
@@ -5375,7 +5375,7 @@ SELECT * FROM TAB ORDER BY ALL SETTINGS enable_order_by_all = 0;
 If enabled, server will ignore all DROP table queries with specified probability (for Memory and JOIN engines it will replcase DROP to TRUNCATE). Used for testing purposes
 )", 0) \
     M(Bool, traverse_shadow_remote_data_paths, false, R"(
-Traverse shadow directory when query system.remote_data_paths
+Traverse frozen data(shadow directory) in addition to actual table data when query system.remote_data_paths
```
Comment (kssenii) [inline src/Core/Settings.cpp:5378]:
```suggestion
Traverse frozen data (shadow directory) in addition to actual table data when query system.remote_data_paths
```

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> AST fuzzer (asan)

https://github.com/ClickHouse/ClickHouse/issues/56640

**Comment by aalexfvk** [comment]
@kssenii Hello! Can we bring all backports to merging. Is it needed any assistance from my side ?

**Comment by kssenii** [comment]
Hi @aalexfvk, I will do soon.

**Pull request merged**